xskillscore/core/np_deterministic.py

```diff
@@ -87,6 +89,77 @@ def _pearson_r(a, b, weights, axis, skipna):
     return res
 
 
+def _spearman_r(a, b, weights, axis, skipna):
+    """
+    ndarray implementation of scipy.stats.spearmanr.
+
+    Parameters
+    ----------
+    a : ndarray
+        Input array.
+    b : ndarray
+        Input array.
+    axis : int
+        The axis to apply the correlation along.
+    weights : ndarray
+        Input array.
+    skipna : bool
+        If True, skip NaNs when computing function.
+
+    Returns
+    -------
+    res : ndarray
+        Spearmanr's correlation coefficient.
+
+    See Also
+    --------
+    scipy.stats.spearmanr
+
+    """
+    if skipna:
+        rankfunc = bn.rankdata
+    else:
+        rankfunc = bn.nanrankdata
```

Comment:
Is this supposed to be under `if skipna`? Also curious to see if this works since the `bottleneck` documentation suggests this just returns nans as nans. Maybe you have to do `nanrankdata` and then some sort of `dropna` function before feeding it into `pearson_r`. You'll figure out in testing I guess